### ISSUE TITLE
Documentation request: printing an info message with location information

### ISSUE THREAD

**Issue by cpitclaudel** [sphinx-doc/sphinx#4976]
### Problem

It's currently fairly hard for extensions to print info messages with location information.

#### Procedure to reproduce the problem

Here's an example of code that one might expect to work:

```
            document.reporter.report_level = docutils.utils.Reporter.INFO_LEVEL
            node = document.reporter.info(msg, line=self.lineno)
```

… but Sphinx strips the location information from the message printed by document.reporter (why does is do that, btw?).

One might also expect this to work:

```
            getLogger(__name__).info(msg, location=(document.reporter.source, self.lineno))
````

But there, too, Sphinx doesn't print location information.  The only way I found is this:

```
            node = document.reporter.info(msg, line=self.lineno)
            getLogger(__name__).info(node.astext())
```

The logging.warning function has a convenient `location` parameter; could the `logging.info` accept such an argument too? Or is there a better way to show info messages with location information? 

Sorry if I missed something.

**Comment by tk0miya** [comment]
>… but Sphinx strips the location information from the message printed by document.reporter (why does is do that, btw?).

So far, Sphinx has suppressed INFO level messages from docutils. Therefore there are no special handler for them.

I don't know why. I guess that was came from the default setting of docutils.

>The logging.warning function has a convenient location parameter; could the logging.info accept such an argument too? Or is there a better way to show info messages with location information?

No, `info()` helper of `sphinx.util.logging` does not support `location` argument. In Sphinx, INFO level messages are used for letting users know progress of build. So location is not necessary.

AFAIK, there are no better way to do that. I'd like to ask you why not `warning()`? Please let me know your situation in detail.

There are no reason to omit `location` argument in `info()`. Simply it was not requested by anybody. So we can add it if it is reasonable and necessary.

**Comment by cpitclaudel** [comment]
Thanks for the detailed response.

> So far, Sphinx has suppressed INFO level messages from docutils.

Indeed, thanks for confirming that.
(For my own future reference: that can be changed using docutils.conf; see https://github.com/sphinx-doc/sphinx/issues/149).

> AFAIK, there are no better way to do that. I'd like to ask you why not warning()? Please let me know your situation in detail.

The tl;dr is that I like to build with `-W`, treating all warning as errors.  In more details, here's the use case:

We recently ported the documentation of the Coq proof assistant to Sphinx.  It was a multiple-days, multiple people effort (the original manual was about 600 pages of LaTeX), and although things went smoothly there are still many issues to correct.

We've recently finished fixing all warnings, so we now build with `-W` every time, to prevent warnings from reoccurring.  But there are still many non-optimal things in the manual.  For example, we have a `.. errormessage::` directive that takes an argument and a body.

```rst
.. errormessage:: Not unifiable: @x and @y

   Raised when `@x` and `@y` are not beta-iota-zeta-convertible.
```

Some people working on the manual wrote this instead:

```rst
.. errormessage:: Not unifiable: @x and @y

Raised when `@x` and `@y` are not beta-iota-zeta-convertible.
```

… which works, because the body isn't mandatory.


There are many things like this, so we're writing checkers for each of them that produce info messages.  These messages are controlled by multiple config options in `conf.py`.

It's a pain to have to add and remove `-W` when building with these flags.

Hopefully this makes sense!

**Comment by tk0miya** [comment]
Sorry for late response.

Thank you for explanation. I just made #5118. Before merging it, I'd like to ask you the `location` option for INFO messages should be permanently? If not, you can patch `sphinx.util.logging` module like following:
```
from sphinx.util.logging import WarningLogRecordTranslator


class InfoLogRecordTranslator(WarningLogRecordTranslator):
    def filter(self, record):
        if not hasattr(record, 'location'):
            return True

        return super(InfoLogRecordTranslator, self).filter(record)


def setup(app):
    import logging
    from sphinx.util.logging import NAMESPACE, NewLineStreamHandler
    logger = logging.getLogger(NAMESPACE)
    for handler in logger.handlers:
        if isinstance(handler, NewLineStreamHandler):  # info handler!
            handler.addFilter(InfoLogRecordTranslator(app))
```

I know this is very hacky and temporary workaround. I don't opposite to add it sphinx-core. So please let me know your opinion.

Thanks,

**Comment by tk0miya** [comment]
Fixed by #5118.
Thanks,

**Comment by cpitclaudel** [comment]
Great, thanks! Sorry for not being very responsive.
